scipy/stats/mstats_basic.py

```diff
@@ -575,14 +595,26 @@ def _kendall_p_exact(n, c):
             if j <= c:
                 new[j:] -= new[:c+1-j]
         prob = np.sum(new)
+        p_mass_at_c = new[-1]/2
 
-    return np.clip(prob, 0, 1)
+    if alternative != 'two-sided':
+        # if the alternative hypothesis and alternative agree,
+        # one-sided p-value is half the two-sided p-value
+        if in_right_tail == alternative_greater:
+            prob /= 2
+        else:
+            prob = 1 - prob/2 + p_mass_at_c
```

Comment:
For discrete statistics, the other one-sided p-value is not just the complement; we also have to include the probability mass _at_ the observed value of the statistic.